x-pack/plugins/enterprise_search/public/applications/workplace_search/types.ts

```diff
@@ -181,6 +181,13 @@ export interface IndexingConfig {
   schedule: IndexingSchedule;
 }
 
+// GitHub apps only
+interface Secret {
```

Comment:
NIT: Since this is the global shared types file, can we name this `AppSecret` or something a bit more descriptive? Not sure what use-case we might have in the future for another "secret" but would prefer this name to be a bit more descriptive for future devs. We could then lose the comment, as I am sure we'll have more source types in the future that use something like this.